src/components/keyboard-nav.js

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash/fp'
+import { useRef } from 'react'
+import * as Utils from 'src/libs/utils'
+
+
+export const withArrowKeyNavigation = ({ onKeyDown, children }) => {
+  const count = children.length
+  const refs = _.map(useRef, _.range(0, count))
```

Comment:
I think the Right Thing To Do here is to return a div with a ref attached to it, and use that ref to access its children, which can then be indexed into; we'd just have to support passing props to the wrapper div.

Reply:
I think I'm following. I'd keep a single ref to the wrapper div, which would returns a raw 'HTMLElement', and then access the children just using the DOM's `children` property?

When I first tried just using `document.querySelector` to find the element and call `focus()` on it directly, I could find the element but the focus didn't seem to transfer. So I thought maybe React doesn't let you fully break out into the DOM to manage focus state, or at least you're not supposed to. But I don't see why it shouldn't work. I'll give it a try.

Reply:
Hmm. I'm trying to attach the keydown event to the wrapper div rather than each individual child. But then I have to figure out which child triggered the event, and the actual Clickable element might be several levels down in the child. I could recursively scan the entire DOM tree below the wrapper element until I find it, but that doesn't seem like the React way to do things...

Reply:
Figured it out!  Addressed in #2512 (I'll close this PR and use those ones instead)